### PR TITLE
Add rules from eslint's recommended set that triggered good lints

### PR DESCRIPTION
**Pull request by jakebailey**
Out of interest, I enabled eslint/tseslint's recommended sets. These rules were ones that fired in our repo with standout results that I believe are good changes in the repo.

### CODE REVIEW

**Comment by jakebailey** [comment]
@typescript-bot perf test this pretty please

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -9198,7 +9198,7 @@ namespace ts {
                 if (container && (container.kind === SyntaxKind.Constructor || isJSConstructor(container))) {
                     return container as ConstructorDeclaration;
                 }
-            };
```
Comment (sandersn) [inline src/compiler/checker.ts:9201]: ❤️ semicolons

File: src/compiler/scanner.ts [hunk 2]
```diff
@@ -359,7 +359,7 @@ namespace ts {
 
     /* @internal */
     export function computeLineStarts(text: string): number[] {
-        const result: number[] = new Array();
+        const result: number[] = [];
```
Comment (sandersn) [inline src/compiler/scanner.ts:362]: Maybe there was some reason for using `new Array` here? But performance doesn't seem to have changed.

**Review by sandersn — APPROVED** [review]

File: src/compiler/binder.ts [hunk 3]
```diff
@@ -3041,8 +3041,11 @@ namespace ts {
                 return;
             }
             const rootExpr = getLeftmostAccessExpression(node.left);
-            if (isIdentifier(rootExpr) && lookupSymbolForName(container, rootExpr.escapedText)!?.flags & SymbolFlags.Alias) {
```
[the remark this answers is not in the document]
Comment (andrewbranch) [reply src/compiler/binder.ts:3044]: I would have written it like `?.flags!` though

File: src/compiler/scanner.ts (hunk 2 above)
Comment (jakebailey) [reply src/compiler/scanner.ts:364]: I don't think so. It's been there since the first commit of the TypeScript compiler.

**Pull request merged**